examples/ITK_Example20_PixelTypes.ipynb

```diff
@@ -0,0 +1,325 @@
+    "for dtype in np_dtypes:\n",
+    "    image = create_random_image(dtype)\n",
+    "    result_image, result_transform_parameters = itk.elastix_registration_method(\n",
+    "    image, image,\n",
+    "    parameter_object=parameter_object,\n",
+    "    log_to_console=False)\n",
+    "    \n",
+    "    print(f\"{itk.template(image)[1][0]}\".ljust(indent) + f\"{itk.template(result_image)[1][0]}\".ljust(indent) + f\"{dtype}\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "1cb169fe",
+   "metadata": {},
+   "source": [
+    "### Transformix pixel types"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "7f67cfa3",
+   "metadata": {},
+   "source": [
+    "The previous example used elastix, but the same functionality works also for transformix"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "0548b67e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Input images type             Output image type             Corresponding numpy dtype\n",
+      "<itkCType unsigned char>      <itkCType unsigned char>      <class 'numpy.uint8'>\n",
+      "<itkCType signed short>       <itkCType signed short>       <class 'numpy.int16'>\n",
+      "<itkCType unsigned short>     <itkCType unsigned short>     <class 'numpy.uint16'>\n",
+      "<itkCType float>              <itkCType float>              <class 'numpy.float32'>\n",
+      "<itkCType double>             <itkCType double>             <class 'numpy.float64'>\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"Input images type\".ljust(indent) + \"Output image type\".ljust(indent) + \"Corresponding numpy dtype\")\n",
+    "for dtype in np_dtypes:\n",
+    "    image = create_random_image(dtype)\n",
+    "    transformed_image = itk.transformix_filter(\n",
+    "    image,\n",
+    "    transform_parameter_object=result_transform_parameters,\n",
+    "    log_to_console=False)\n",
+    "    \n",
+    "    print(f\"{itk.template(image)[1][0]}\".ljust(indent) + f\"{itk.template(transformed_image)[1][0]}\".ljust(indent) + f\"{dtype}\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "6935081a",
+   "metadata": {},
+   "source": [
+    "### Internal pixel types"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "59c0389f",
+   "metadata": {},
+   "source": [
+    "What happens as under the hood is that elastix/transformix will [cast](https://itk.org/Doxygen/html/classitk_1_1CastImageFilter.html) the input image(s) to either `float` (default) or `short` depending on the value of the parameters `FixedInternalPixelType` and `MovingInternalPixelType` specified in the parameter object. When the registration or tranformation is over, the image will be converted back (cast) to original pixel type before it is output to the user."
```

Comment:
Nice description. As a minor style suggestion: I would slightly shorten the first sentence and then add another sentence, like:

```
Under the hood, elastix/transformix will cast the pixels of the input image(s) to an internal pixel type.
By default, the internal pixel type is `float`, but for 3D and 4D images, it can be `short`, according to the value of the parameters `FixedInternalPixelType` and `MovingInternalPixelType` specified in the parameter object. 
```

Little typo: "tranformation"

Very _very_ minor: "When the registration or transformation is over" makes me feel a bit sad.  I heard too many sad songs like "I Know It's Over" and "It Ain't Over 'Til It's Over"...  Maybe replace "over" with "finished"?  😸